x/cork/keeper/keeper.go

```diff
@@ -0,0 +1,272 @@
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", "x/"+types.ModuleName)
+}
+
+///////////////////////
+// MsgSubmitCork //
+///////////////////////
+
+// SetCork sets the prevote for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) SetCork(ctx sdk.Context, val sdk.ValAddress, cork types.Cork) {
+	bz := k.cdc.MustMarshal(&cork)
+	ctx.KVStore(k.storeKey).Set(types.GetCorkForValidatorAddressKey(val, common.HexToAddress(cork.Address)), bz)
+}
+
+// GetCork gets the prevote for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) GetCork(ctx sdk.Context, val sdk.ValAddress, cel common.Address) (types.Cork, bool) {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := store.Get(types.GetCorkForValidatorAddressKey(val, cel))
+	if len(bz) == 0 {
+		return types.Cork{}, false
+	}
+
+	var cork types.Cork
+	k.cdc.MustUnmarshal(bz, &cork)
+	return cork, true
+}
+
+// DeleteCork deletes the prevote for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) DeleteCork(ctx sdk.Context, val sdk.ValAddress, cel common.Address) {
+	ctx.KVStore(k.storeKey).Delete(types.GetCorkForValidatorAddressKey(val, cel))
+}
+
+// HasCorkForContract gets the prevote for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) HasCorkForContract(ctx sdk.Context, val sdk.ValAddress, contract common.Address) bool {
+	return ctx.KVStore(k.storeKey).Has(types.GetCorkForValidatorAddressKey(val, contract))
+}
+
+// HasCork gets the existence of any commit for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) HasCork(ctx sdk.Context, val sdk.ValAddress) bool {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.GetCorkValidatorKeyPrefix(val))
+	iter := store.Iterator(nil, nil)
+	defer iter.Close()
+
+	return iter.Valid()
+}
+
+// IterateCorks iterates over all votes in the store
+func (k Keeper) IterateCorks(ctx sdk.Context, handler func(val sdk.ValAddress, cel common.Address, cork types.Cork) (stop bool)) {
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, []byte{types.CorkForAddressKeyPrefix})
+	defer iter.Close()
+	for ; iter.Valid(); iter.Next() {
+		keyPair := bytes.NewBuffer(bytes.TrimPrefix(iter.Key(), []byte{types.CorkForAddressKeyPrefix}))
+		val := sdk.ValAddress(keyPair.Next(20))
+		cel := common.BytesToAddress(keyPair.Bytes())
+
+		var cork types.Cork
+		k.cdc.MustUnmarshal(iter.Value(), &cork)
+		if handler(val, cel, cork) {
+			break
+		}
+	}
+}
+
+// IterateCorkAddresses iterates over all addresses who have committed corks
+func (k Keeper) IterateCorkAddresses(ctx sdk.Context, handler func(addr common.Address) (stop bool)) {
```

Comment:
It looks like this function and `IterateAddressCorks` have the exact same code.

Reply:
And neither are used. Deleted both